src/stream/pending.rs

```diff
@@ -0,0 +1,45 @@
+use std::marker::PhantomData;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use crate::stream::{Stream, DoubleEndedStream, ExactSizeStream, FusedStream};
+
+/// A stream that never returns any items.
+/// 
+/// This stream is created by the [`pending`] function. See its
+/// documentation for more.
+/// 
+/// [`pending`]: fn.pending.html
+#[derive(Debug)]
+pub struct Pending<T> {
+    _marker: PhantomData<T>
+}
+
+/// Creates a stream that never returns any items.
+/// 
+/// The returned stream will always return `Pending` when polled.
```

Comment:
Could we add a usage example here?

Reply:
The reason I haven't added an example here is that the pending stream will never return anything. It is possible to add an example that polls instead of `await`s the stream but I think the behavior of pending stream is straightforward enough that such an example isn't needed. Plus, the [`futures_util::stream::pending`](https://docs.rs/futures-util/0.3.1/futures_util/stream/fn.pending.html) function does not come with an example.

Reply:
> Plus, the futures_util::stream::pending function does not come with an example.

We have an example for [`future::pending`](https://docs.rs/async-std/1.3.0/async_std/future/fn.pending.html) and given `stream::pending` fills much the same role I'd like us to have an example here.

The example we're using there is in combination with a `timeout`, showing the timeout is always hit as the stream will never resolve. That's probably a useful practical use case as well, and showcasing it for stream would be useful as well I think (: